// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless  by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

syntax="proto3";
package ocean_kv.proxy;

option cc_generic_services = true;

message GetRequest {
       string key = 1;
};

message GetResponse {
       uint32 code = 1;
       bytes value = 2;
};

message PutRequest {
       string key = 1;
       bytes value = 2;
};

message PutResponse {
       uint32 code = 1;
};


message DeleteRequest {
       string key = 1;
       bool need_value = 2;
};

message DeleteResponse {
       uint32 code = 1;
       bytes value = 2;
};

service ProxyService {
      rpc Get(GetRequest) returns (GetResponse);
      rpc Put(PutRequest) returns (PutResponse);
      rpc Delete(DeleteRequest) returns (DeleteResponse);
};
